tests/integration.rs

```diff
@@ -0,0 +1,136 @@
+extern crate assert_cmd;
```

Comment:
As it were I've been working on a sort of integration testing framework for exactly this sort of thing in the app framework I've been hoping to move this tool to. I'll accept this for now though... I can port it over!

Reply:
Aha - I'll be interested in what you come up with. I had not tested mains in rust yet and `assert_cmd` was the first helper I found.

Reply:
The approach is a bit different in that it's a framework-integrated testing approach that pulls the application in as a Rust library rather that invoking it as a subprocess, which while not providing a full from-the-terminal "integration" test does allow it to do a much better job reporting unhandled panics, with the idea that things like terminal interactions can be integration tested at the framework level.

Reply:
Nice.